cmd/check-cluster-profiles-config/main.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	coreapi "k8s.io/api/core/v1"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	configPath string
+}
+
+type profileValidator struct {
+	profiles   api.ClusterProfilesMap
+	kubeClient ctrlruntimeclient.Client
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.configPath, "config-path", "", "Path to the cluster profile config file")
+
+	if err := fs.Parse(os.Args[1:]); err != nil { // todo: should this be args[1:] ?
+		logrus.WithError(err).Fatal("could not parse arguments")
+	}
+
+	return o
+}
+
+func newValidator() *profileValidator {
+	inClusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to load in-cluster config")
+	}
+
+	client, err := ctrlruntimeclient.New(inClusterConfig, ctrlruntimeclient.Options{})
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to create client")
+	}
+
+	return &profileValidator{
+		profiles:   make(api.ClusterProfilesMap),
+		kubeClient: client,
+	}
+}
+
+func main() {
+	logger := logrus.WithField("component", "check-cluster-profiles-config")
+	o := gatherOptions()
+
+	validator := newValidator()
+	if err := validator.loadConfig(o.configPath); err != nil {
+		logger.WithError(err).Fatal("failed to load profiles from config")
+	}
+
+	if err := validator.checkCiSecrets(); err != nil {
+		logger.WithError(err).Fatal("failed to check secrets for cluster profiles")
+	}
+
+	// TODO after they are implemented into the config file, add checks for cluster type and leases
+
+	logger.Info("Cluster profiles successfully checked.")
+}
+
+func (validator *profileValidator) loadConfig(configPath string) error {
+	configContents, err := os.ReadFile(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to read cluster profiles config: %w", err)
+	}
+
+	var profilesList api.ClusterProfilesList
+	if err = yaml.Unmarshal(configContents, &profilesList); err != nil {
+		return fmt.Errorf("failed to unmarshall file %v. Please check that the formatting in the file is correct. Full error: %w", configPath, err)
+	}
+
+	for _, p := range profilesList {
+		// Check if a profile isn't already defined in the config
+		if _, found := validator.profiles[p.Profile]; found {
+			return fmt.Errorf("cluster profile '%v' already exists in the configuration file", p.Profile)
+		}
+		validator.profiles[p.Profile] = p
+	}
+	return nil
+}
+
+// checkCiSecrets verifies that the secret for each cluster profile exists in the ci namespace
+func (validator *profileValidator) checkCiSecrets() error {
```

Comment:
We should assume that if the secret exists in the bootstrap entry it would exist in the cluter as well. I am not sure if that behavior will be changed after the new secret vault in GCP. @danilo-gemoli ?

Reply:
Yes that's correct, once we implement the new secret manager not all the secrets might be synchronized on the cluster. I can't tell when that will happen though.

Reply:
What way should be used then, after the new secret manager is implemented, to check if a secret for a cluster profile exists?

Reply:
Mount the secrets on a pod, run it and check

Reply:
> Mount the secrets on a pod, run it and check

This PR's purpose is to make ci-operator stop doing that.